### DISCUSSION TITLE
Custom dataset

### DISCUSSION THREAD

**Discussion by DRM-Free** [THU-DA-6D-Pose-Group/GDR-Net#62] [Q&A]
Hello,
I would like to use this work in my project but I'm experiencing difficulties extending the linemod dataset with additional classes. Is there a tutorial I can read somewhere on how to do it ?

**Comment by wangg12** [comment]
Hi, I suggest some steps:
* make your dataset re-organized similar to the bop datasets
* add a dataset config file in `ref/`
* implement a dataset file similar to https://github.com/THU-DA-6D-Pose-Group/GDR-Net/blob/main/core/gdrn_modeling/datasets/lm_dataset_d2.py

**Comment by DRM-Free** [comment]
I understood that you use several datasets (lm, lm opengl and lm blender). Should I replicate only lm? Also, I'm unsure how I can "make my dataset re-organized similar to" the bop lm dataset. Isn't there supposed to be some code available for generating lm dataset from object models?

**Reply by wangg12** [reply]
All the datasets share the same intermediate data structure. So if you already have all the annotations, you can load the data and annotations similar to any of implemented datasets, such as https://github.com/THU-DA-6D-Pose-Group/GDR-Net/blob/main/core/gdrn_modeling/datasets/lm_dataset_d2.py.

**Reply by DRM-Free** [reply]
Thank you for your assistance. I had problems before trying to replicate results with my own dataset so I want to be extra careful with the data. I have my own data generation script made with blender but I would be more comfortable using an "official" one. I get that the pbr data can be safely generated from [Blenderproc](https://github.com/DLR-RM/BlenderProc), which is nice. If I understand well, I should be able to replicate the BOP_DATASETS/lm folder with this tool, is that correct?

**Reply by wangg12** [reply]
BlenderProc can generate the data in BOP_DATASETS/lm/train_pbr.

**Comment by DRM-Free** [comment]
I opened a [question](https://github.com/zju3dv/pvnet-rendering/issues/67) on pvnet on how to replicate their dataset with my own 3d model. Turns out they recommend to not use their code but blenderproc instead. Do you recommend I do as they say or is the pvned data necessary?

**Reply by wangg12** [reply]
BlenderProc should be better.

**Reply by DRM-Free** [reply]
I am currently testing blenderproc and I managed to generate some samples. Now, I need to adapt the software in order to generate a dataset composed of only the 3d model of my choosing, keeping other tless 3d models as distractors. I don't know how to do this. I noticed that in your published results you have trained some models on single objects, so maybe you could help with that.

**Reply by wangg12** [reply]
I just used the pbr data from the official BOP challenge for training, so maybe you need to ask the developers of blenderproc for what you want to do.
